lib/http-proxy/passes/web-incoming.js

```diff
@@ -162,16 +162,22 @@ module.exports = {
 
     proxyReq.on('response', function(proxyRes) {
       if(server) { server.emit('proxyRes', proxyRes, req, res); }
-      for(var i=0; i < web_o.length; i++) {
-        if(web_o[i](req, res, proxyRes, options)) { break; }
+      if (!res.headersSent) {
+        for(var i=0; i < web_o.length; i++) {
+          if(web_o[i](req, res, proxyRes, options)) { break; }
```

Comment:
I think this is a better fix: https://github.com/nodejitsu/node-http-proxy/pull/1182